### PR TITLE
TPU CI testing

### PR DESCRIPTION
**Pull request by zcain117**
Run TPU CI testing using CircleCI.

I sent a guide in Slack with steps needed on the owner side for CircleCI and Google Cloud to make this work (setting env vars, creating GKE cluster, and populating the dataset).

### CODE REVIEW

**Comment by zcain117** [comment]
@LysandreJik I'm wondering if CircleCI does not run the pending changes to `.circleci/config.yml` if the changes came from a PR from a forked repo.

When testing on my private repo, I used a branch on the main repo and CircleCI did include the pending changes when running. I've seen lots of differences between branches on main repo and forked repo PRs when it comes to CircleCI and Github Actions.

The action item might be to remake this PR as a branch on the repo rather than a forked PR if you'd like to see the `job` run before submit.

**Comment by zcain117** [comment]
> @LysandreJik I'm wondering if CircleCI does not run the pending changes to `.circleci/config.yml` if the changes came from a PR from a forked repo.
> 
> When testing on my private repo, I used a branch on the main repo and CircleCI did include the pending changes when running. I've seen lots of differences between branches on main repo and forked repo PRs when it comes to CircleCI and Github Actions.
> 
> The action item might be to remake this PR as a branch on the repo rather than a forked PR if you'd like to see the `job` run before submit.

I think the forked repo is irrelevant. I was able to run new CircleCI changes in a similar PR for PyTorch Lightning: https://github.com/PyTorchLightning/pytorch-lightning/pull/2486

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,6 +135,24 @@ jobs:
             - store_artifacts:
                   path: ~/transformers/output.txt
                   destination: test_output.txt
+    run_examples_tpu:
```
Comment (Borda) [inline .circleci/config.yml:138]: @zcain117 you need to add it to the workflow on line 219

Comment (zcain117) [reply .circleci/config.yml:138]: Great find, thank you!

**Comment by zcain117** [comment]
Looks like this rebase went wrong - recreated the change in https://github.com/huggingface/transformers/pull/6158

**Pull request closed without merging**